test/deal-guarantee/post-deal-guarantee.api-test.ts

```diff
@@ -0,0 +1,651 @@
+    const strategy = AUTH.STRATEGY;
+    const randomisedApiKey = valueGenerator.string();
+    const { status, body } = await api.postWithoutAuth(createDealGuaranteeUrl, requestBodyToCreateDealGuarantee, strategy, randomisedApiKey);
+
+    expect(status).toBe(401);
+    expect(body).toStrictEqual({ message: 'Unauthorized', statusCode: 401 });
+  });
+
+  it('returns a 401 response when the API Key is empty', async () => {
+    givenAuthenticationWithTheIdpSucceeds();
+    givenAnyRequestBodyToCreateDealGuaranteeInAcbsSucceeds();
+    const strategy = AUTH.STRATEGY;
+
+    const { status, body } = await api.postWithoutAuth(createDealGuaranteeUrl, requestBodyToCreateDealGuarantee, strategy, '');
+
+    expect(status).toBe(401);
+    expect(body).toStrictEqual({ message: 'Unauthorized', statusCode: 401 });
+  });
+
+  it('returns a 404 response if ACBS responds with a 400 response that is a string containing "The deal not found"', async () => {
+    givenAuthenticationWithTheIdpSucceeds();
+    requestToCreateDealGuarantee().reply(400, 'The deal not found or user does not have access');
+
+    const { status, body } = await api.post(createDealGuaranteeUrl, requestBodyToCreateDealGuarantee);
+
+    expect(status).toBe(404);
+    expect(body).toStrictEqual({ message: 'Not found', statusCode: 404 });
+  });
+
+  it('returns a 400 response if ACBS responds with a 400 response that is not a string', async () => {
+    givenAuthenticationWithTheIdpSucceeds();
+    const acbsErrorMessage = { Message: 'error message' };
+    requestToCreateDealGuarantee().reply(400, acbsErrorMessage);
+
+    const { status, body } = await api.post(createDealGuaranteeUrl, requestBodyToCreateDealGuarantee);
+
+    expect(status).toBe(400);
+    expect(body).toStrictEqual({ message: 'Bad request', error: JSON.stringify(acbsErrorMessage), statusCode: 400 });
+  });
+
+  it('returns a 400 response if ACBS responds with a 400 response that is a string that does not contain "The deal not found"', async () => {
+    givenAuthenticationWithTheIdpSucceeds();
+    const acbsErrorMessage = 'ACBS error message';
+    requestToCreateDealGuarantee().reply(400, acbsErrorMessage);
+
+    const { status, body } = await api.post(createDealGuaranteeUrl, requestBodyToCreateDealGuarantee);
+
+    expect(status).toBe(400);
+    expect(body).toStrictEqual({ message: 'Bad request', error: acbsErrorMessage, statusCode: 400 });
+  });
+
+  it('returns a 500 response if ACBS responds with an error code that is not 400"', async () => {
+    givenAuthenticationWithTheIdpSucceeds();
+    requestToCreateDealGuarantee().reply(401, 'Unauthorized');
+
+    const { status, body } = await api.post(createDealGuaranteeUrl, requestBodyToCreateDealGuarantee);
+
+    expect(status).toBe(500);
+    expect(body).toStrictEqual({ message: 'Internal server error', statusCode: 500 });
+  });
+
+  // TODO APIM-73: Should we test defaults are set in the API tests also?
```

Comment:
These API tests don't check that:
- effective date is replaced if it's earlier than today
- maximumLiability is rounded to 2dp
- guarantorParty and guaranteeTypeCode have defaults if they are not provided
but these are tested in unit tests

Should we test them in these api tests as well or would that be overkill?